src/gateway.c

```diff
@@ -30,6 +33,14 @@ void gateway__init(struct gateway *g,
 	g->barrier.cb = NULL;
 	g->barrier.leader = NULL;
 	g->protocol = DQLITE_PROTOCOL_VERSION;
+	g->random_state = 0;
+	/* For seeding the ID generator, there are no security concerns, so read from
+	 * /dev/urandom instead of calling getrandom(2) to avoid potential blocking. */
+	urandom = open("/dev/urandom", O_RDONLY);
+	assert(urandom != -1);
```

Comment:
Now that I think of it, shouldn't we just initialize the random state once, when the server starts? Now we have to do it for every connection and in real-life a lot of connections get set up and broken down.